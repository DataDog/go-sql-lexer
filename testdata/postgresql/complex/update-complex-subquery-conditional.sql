UPDATE 
  products p
SET 
  price = CASE 
    WHEN p.stock < 10 THEN p.price * 1.10
    WHEN p.stock BETWEEN 10 AND 50 THEN p.price
    ELSE p.price * 0.90
  END,
  last_updated = NOW()
FROM (
  SELECT 
    product_id, 
    SUM(quantity) AS stock
  FROM 
    inventory
  GROUP BY 
    product_id
) AS sub
WHERE 
  sub.product_id = p.id;
