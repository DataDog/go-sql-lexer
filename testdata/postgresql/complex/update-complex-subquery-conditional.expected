[
    {
        "expected": "UPDATE products p SET price = CASE WHEN p.stock < ? THEN p.price * ? WHEN p.stock BETWEEN ? AND ? THEN p.price ELSE p.price * ? END, last_updated = NOW ( ) FROM ( SELECT product_id, SUM ( quantity ) FROM inventory GROUP BY product_id ) WHERE sub.product_id = p.id",
        "statement_metadata": {
            "size": 29,
            "tables": [
                "products",
                "inventory"
            ],
            "commands": [
                "UPDATE",
                "SELECT"
            ],
            "comments": [],
            "procedures": []
        }
    }
]